bugbug/tools/code_review.py

```diff
@@ -129,26 +286,87 @@ def get_review_request_by_id(self, revision_id: int) -> ReviewRequest:
         assert len(revisions) == 1
         return ReviewRequest(revisions[0]["fields"]["diffID"])
 
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(7),
+        wait=tenacity.wait_exponential(multiplier=1, min=16, max=64),
+        reraise=True,
+    )
     def get_patch_by_id(self, patch_id: int) -> Patch:
         assert phabricator.PHABRICATOR_API is not None
         raw_diff = phabricator.PHABRICATOR_API.load_raw_diff(int(patch_id))
         return Patch(raw_diff)
 
+    def _get_revisions_count(self):
+        return sum(1 for _ in phabricator.get_revisions())
+
+    def get_all_inline_comments(
+        self, comment_filter
+    ) -> Iterable[tuple[int, list[InlineComment]]]:
+        db.download(phabricator.REVISIONS_DB)
+
+        for revision in tqdm(
+            phabricator.get_revisions(), total=self._get_revisions_count()
+        ):
+            diff_comments: dict[int, list[InlineComment]] = defaultdict(list)
+
+            for transaction in revision["transactions"]:
+                if transaction["type"] != "inline":
+                    continue
+
+                # Ignore replies
+                if transaction["fields"]["replyToCommentPHID"] is not None:
+                    continue
+
+                # Ignore bot comments
+                if transaction["authorPHID"] == "PHID-USER-cje4weq32o3xyuegalpj":
+                    continue
+
+                if len(transaction["comments"]) != 1:
+                    print(transaction)
+                    # raise Exception("Unexpected, need to look into it")
```

Comment:
Log warning?

Reply:
Done in 7aef774a887c24eb48e19346126dd72c9ea37a9b